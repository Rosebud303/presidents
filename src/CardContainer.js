import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from './Card'
import './CardContainer.css'


export class CardContainer extends Component {
  render() {
    const displayCards = this.props.presidents.map( president => {
      return (
        <Card {...president} key={president.number} />
      )
    })
    return (
      <div className='container'>
        {displayCards}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  presidents: state.presidents
})

export default connect(mapStateToProps)(CardContainer)