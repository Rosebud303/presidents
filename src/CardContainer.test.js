import React from 'react'
import { CardContainer, mapStateToProps } from './CardContainer'
import { shallow } from 'enzyme'

describe('CardContainer', () => {
  describe('Component', () => {
    // let wrapper;

    // beforeEach(() => {
    //   wrapper = shallow(
    //     <CardContainer />
    //   )
    // });

    // it('should match the snapshot', () => {
    //   expect(wrapper).toMatchSnapshot()
    // })
  })
  
  describe('mapStateToProps', () => {
    it('should have a mapped state prop', () => {
      let mockstate = {
        isLoading: false,
        error: '',
        presidents: []
      }
      let expected = {
        presidents: []
      }
      const result = mapStateToProps(mockstate)
      expect(result).toEqual(expected)
    })
  });
})
