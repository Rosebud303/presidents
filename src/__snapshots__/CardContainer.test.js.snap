// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Component should match the snapshot 1`] = `undefined`;
